--- rezound-0.8.3beta/src/frontend_fox/CProgressDialog.cpp.orig	2003-06-13 19:01:20.000000000 +0200
+++ rezound-0.8.3beta/src/frontend_fox/CProgressDialog.cpp	2003-10-24 08:21:59.308405704 +0200
@@ -67,7 +67,7 @@
 {
 	progressBar->setProgress(progress);
 	if(cancelButton)
-		getApp()->runWhileEvents(cancelButton); // give cancel button an oppertunity to be clicked
+		getApp()->runModalWhileEvents(cancelButton); // give cancel button an oppertunity to be clicked
 }
 
 void CProgressDialog::show(FXuint placement)
--- rezound-0.8.3beta/src/frontend_fox/rememberShow.cpp.orig	2003-06-27 01:18:36.000000000 +0200
+++ rezound-0.8.3beta/src/frontend_fox/rememberShow.cpp	2003-10-24 08:52:03.576115320 +0200
@@ -85,12 +85,12 @@
 	new FXLabel(testWindow,"Testing 1 2 3...");
 	testWindow->create();
 	testWindow->show();
-	testWindow->getApp()->runWhileEvents(testWindow); // wait for it to show
+	testWindow->getApp()->runModalWhileEvents(testWindow); // wait for it to show
 
 	testWindow->move(100,100);
 	testWindow->hide();
 	testWindow->show();
-	testWindow->getApp()->runWhileEvents(testWindow); // wait for it to show again
+	testWindow->getApp()->runModalWhileEvents(testWindow); // wait for it to show again
 
 	decorWidth=testWindow->getX()-100;
 	decorHeight=testWindow->getY()-100;
