--- rezound-0.9.0beta/src/frontend_fox/custom_cursors.h.orig	2002-12-25 10:03:28.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/custom_cursors.h	2004-01-15 22:55:45.472211144 +0100
@@ -38,9 +38,9 @@
 };
 
 
-static void *bytesToBits(const char *bytes,int bytesLength)
+static FXuchar *bytesToBits(const char *bytes,int bytesLength)
 {
-	char *bits=new char[(bytesLength/8)+1];
+	FXuchar *bits=new FXuchar[(bytesLength/8)+1];
 	for(int t=0;t<bytesLength/8+1;t++)
 		bits[t]=0; 
 	for(int t=0;t<bytesLength;t++)
--- rezound-0.9.0beta/src/frontend_fox/CActionParamDialog.cpp.orig	2003-10-27 22:09:14.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/CActionParamDialog.cpp	2004-02-20 00:49:45.000000000 +0100
@@ -910,7 +910,8 @@
 			{
 				// native preset stuff
 				FXPacker *listFrame=new FXPacker(presetsFrame,FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH | LAYOUT_FILL_Y, 0,0,210,0, 0,0,0,0, 0,0); // had to do this because FXList won't take that frame style
-					nativePresetList=new FXList(listFrame,4,this,ID_NATIVE_PRESET_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y);
+					nativePresetList=new FXList(listFrame,this,ID_NATIVE_PRESET_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y);
+					nativePresetList->setNumVisible(4);
 				new FXButton(presetsFrame,"&Use\tOr Double-Click an Item in the List",NULL,this,ID_NATIVE_PRESET_BUTTON);
 			}
 		}
@@ -922,7 +923,8 @@
 
 		// user preset stuff
 		FXPacker *listFrame=new FXPacker(presetsFrame,FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH | LAYOUT_FILL_Y, 0,0,210,0, 0,0,0,0, 0,0); // had to do this because FXList won't take that frame style
-			userPresetList=new FXList(listFrame,4,this,ID_USER_PRESET_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y);
+			userPresetList=new FXList(listFrame,this,ID_USER_PRESET_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y);
+			userPresetList->setNumVisible(4);
 		FXPacker *buttonGroup=new FXVerticalFrame(presetsFrame);
 			new FXButton(buttonGroup,_("&Use\tOr Double-Click an Item in the List"),NULL,this,ID_USER_PRESET_USE_BUTTON,BUTTON_NORMAL|LAYOUT_FILL_X);
 			new FXButton(buttonGroup,_("&Save"),NULL,this,ID_USER_PRESET_SAVE_BUTTON,BUTTON_NORMAL|LAYOUT_FILL_X);
--- rezound-0.9.0beta/src/frontend_fox/CCueListDialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CCueListDialog.cpp	2004-02-20 00:51:34.000000000 +0100
@@ -68,7 +68,7 @@
 	contents(new FXVerticalFrame(this,LAYOUT_FILL_X|LAYOUT_FILL_Y, 0,0,0,0, 0,0,0,0, 0,0)),
 		upperFrame(new FXVerticalFrame(contents,FRAME_RAISED|FRAME_THICK | LAYOUT_FILL_X|LAYOUT_FILL_Y, 0,0,0,0, 5,5,5,5)),
 			cueListFrame(new FXPacker(upperFrame,LAYOUT_FILL_X|LAYOUT_FILL_Y | FRAME_SUNKEN|FRAME_THICK, 0,0,0,0, 0,0,0,0, 0,0)),
-				cueList(new FXList(cueListFrame,0,this,ID_CUE_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y)),
+				cueList(new FXList(cueListFrame,this,ID_CUE_LIST,LIST_BROWSESELECT | LAYOUT_FILL_X|LAYOUT_FILL_Y)),
 		lowerFrame(new FXHorizontalFrame(contents,FRAME_RAISED|FRAME_THICK | LAYOUT_FILL_X|LAYOUT_FIX_HEIGHT, 0,0,0,65)),
 			buttonPacker(new FXHorizontalFrame(lowerFrame,LAYOUT_CENTER_X|LAYOUT_CENTER_Y, 0,0,0,0, 0,0,0,0, 12)),
 				closeButton(new FXButton(buttonPacker,_("&Close"),FOXIcons->GreenCheck1,this,ID_CLOSE_BUTTON,FRAME_RAISED|FRAME_THICK | JUSTIFY_NORMAL | ICON_ABOVE_TEXT | LAYOUT_FIX_WIDTH, 0,0,60,0, 2,2,2,2)),
--- rezound-0.9.0beta/src/frontend_fox/CMainWindow.cpp.orig	2003-10-22 07:15:17.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CMainWindow.cpp	2004-02-20 00:55:50.000000000 +0100
@@ -252,14 +252,16 @@
 		drawVerticalCuePositionsButton=new FXCheckButton(miscControlsFrame,_("Draw Vertical Cue Positions"),this,ID_DRAW_VERTICAL_CUE_POSITIONS_TOGGLE);
 		drawVerticalCuePositionsButton->setPadLeft(0); drawVerticalCuePositionsButton->setPadRight(0); drawVerticalCuePositionsButton->setPadTop(0); drawVerticalCuePositionsButton->setPadBottom(0);
 		t=new FXHorizontalFrame(miscControlsFrame,0, 0,0,0,0, 0,0,0,0);
-			crossfadeEdgesComboBox=new FXComboBox(t,8,3, this,ID_CROSSFADE_EDGES_COMBOBOX, FRAME_SUNKEN|FRAME_THICK | COMBOBOX_NORMAL|COMBOBOX_STATIC | LAYOUT_CENTER_Y);
+			crossfadeEdgesComboBox=new FXComboBox(t,8, this,ID_CROSSFADE_EDGES_COMBOBOX, FRAME_SUNKEN|FRAME_THICK | COMBOBOX_NORMAL|COMBOBOX_STATIC | LAYOUT_CENTER_Y);
+				crossfadeEdgesComboBox->setNumVisible(3);
 				crossfadeEdgesComboBox->setTipText(_("After Most Actions a Crossfade can be Performed at the Start and Stop \nPositions to Give a Smoother Transition in to and out of the Modified Selection"));
 				crossfadeEdgesComboBox->appendItem(_("No Crossfade"));
 				crossfadeEdgesComboBox->appendItem(_("Crossfade Inner Edges"));
 				crossfadeEdgesComboBox->appendItem(_("Crossfade Outer Edges"));
 				crossfadeEdgesComboBox->setCurrentItem(0);
 			new FXButton(t,FXString("...\t")+_("Change Crossfade Times"),NULL,this,ID_CROSSFADE_EDGES_SETTINGS_BUTTON, BUTTON_NORMAL & ~FRAME_THICK);
-		clipboardComboBox=new FXComboBox(miscControlsFrame,8,8, this,ID_CLIPBOARD_COMBOBOX, FRAME_SUNKEN|FRAME_THICK | COMBOBOX_NORMAL|COMBOBOX_STATIC);
+		clipboardComboBox=new FXComboBox(miscControlsFrame,8, this,ID_CLIPBOARD_COMBOBOX, FRAME_SUNKEN|FRAME_THICK | COMBOBOX_NORMAL|COMBOBOX_STATIC);
+			clipboardComboBox->setNumVisible(8);
 
 	new FXVerticalSeparator(s);
 
--- rezound-0.9.0beta/src/frontend_fox/CNewSoundDialog.cpp.orig	2003-10-28 07:09:37.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/CNewSoundDialog.cpp	2004-02-20 01:07:57.000000000 +0100
@@ -60,20 +60,24 @@
 	matrix(new FXMatrix(getFrame(),2,MATRIX_BY_COLUMNS|LAYOUT_CENTER_X)),
 
 		channelsLabel(new FXLabel(matrix,_("Channels:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT)),
-		channelsComboBox(new FXComboBox(matrix,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC)),
+		channelsComboBox(new FXComboBox(matrix,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC)),
 
 		sampleRateLabel(new FXLabel(matrix,_("Sample Rate:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT)),
-		sampleRateComboBox(new FXComboBox(matrix,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK)),
+		sampleRateComboBox(new FXComboBox(matrix,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK)),
 
 		lengthLabel(new FXLabel(matrix,_("Length:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT)),
 		lengthFrame(new FXHorizontalFrame(matrix,LAYOUT_CENTER_X,0,0,0,0, 0,0,0,0, 0,0)),
-			lengthComboBox(new FXComboBox(lengthFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK)),
+			lengthComboBox(new FXComboBox(lengthFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK)),
 			lengthUnitsLabel(new FXLabel(lengthFrame,_("second(s)"))),
 
 
 	rememberAsDefault(new FXCheckButton(getFrame(),_("Remember as Default"),NULL,0,CHECKBUTTON_NORMAL|LAYOUT_BOTTOM|LAYOUT_CENTER_X)),
 	separator(new FXHorizontalSeparator(getFrame(),SEPARATOR_GROOVE|LAYOUT_FILL_X|LAYOUT_BOTTOM))
 {
+	channelsComboBox->setNumVisible(8);
+	sampleRateComboBox->setNumVisible(8);
+	lengthComboBox->setNumVisible(8);
+
 	if(!ASoundTranslator::findRawTranslator()) // hide if we can't handle it
 		rawFormatFrame->hide();
 
--- rezound-0.9.0beta/src/frontend_fox/CPasteChannelsDialog.cpp.orig	2003-10-24 08:51:09.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CPasteChannelsDialog.cpp	2004-02-20 01:10:57.524114688 +0100
@@ -88,7 +88,8 @@
 
 		repeatTypeSwitcher->setCurrent(0);
 
-	repeatTypeComboBox=new FXComboBox(repeatFrame,0,2,this,ID_REPEAT_TYPE_COMBOBOX,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN|FRAME_THICK | LAYOUT_CENTER_X);
+	repeatTypeComboBox=new FXComboBox(repeatFrame,0,this,ID_REPEAT_TYPE_COMBOBOX,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN|FRAME_THICK | LAYOUT_CENTER_X);
+		repeatTypeComboBox->setNumVisible(2);
 		repeatTypeComboBox->appendItem(_("Repeat X Times"));
 		repeatTypeComboBox->appendItem(_("Repeat for X Seconds"));
 		repeatTypeComboBox->setCurrentItem(0);
@@ -104,7 +105,8 @@
 		new FXButton(buttonPacker,_("Clear"),NULL,this,ID_CLEAR_BUTTON,BUTTON_NORMAL|LAYOUT_CENTER_Y);
 		mixTypeFrame=new FXHorizontalFrame(buttonPacker,LAYOUT_CENTER_X|LAYOUT_CENTER_Y);
 			new FXLabel(mixTypeFrame,_("Mix Type:"),NULL,LAYOUT_CENTER_Y);
-			mixTypeComboBox=new FXComboBox(mixTypeFrame,16,4,NULL,0, COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK | COMBOBOX_STATIC | LAYOUT_CENTER_Y);
+			mixTypeComboBox=new FXComboBox(mixTypeFrame,16,NULL,0, COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK | COMBOBOX_STATIC | LAYOUT_CENTER_Y);
+			mixTypeComboBox->setNumVisible(4);
 			mixTypeComboBox->appendItem(_("Add"),(void *)mmAdd);
 			mixTypeComboBox->appendItem(_("Subtract"),(void *)mmSubtract);
 			mixTypeComboBox->appendItem(_("Multiply"),(void *)mmMultiply);
--- rezound-0.9.0beta/src/frontend_fox/CRawDialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CRawDialog.cpp	2004-02-20 01:17:22.600574168 +0100
@@ -49,13 +49,15 @@
 	FXComboBox *combo;
 
 	new FXLabel(main,_("Channels:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT);
-	combo=channelsCountComboBox=new FXComboBox(main,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+	combo=channelsCountComboBox=new FXComboBox(main,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+		combo->setNumVisible(8);
 		for(unsigned t=1;t<=MAX_CHANNELS;t++)
 			combo->appendItem(istring(t).c_str());
 		combo->setCurrentItem(1); // stereo
 
 	new FXLabel(main,_("Sample Rate:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT);
-	combo=sampleRateComboBox=new FXComboBox(main,10,9,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+	combo=sampleRateComboBox=new FXComboBox(main,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+		combo->setNumVisible(9);
 		combo->appendItem("4000");
 		combo->appendItem("8000");
 		combo->appendItem("11025");
@@ -68,7 +70,8 @@
 		combo->setCurrentItem(5);
 
 	new FXLabel(main,_("Sample Format:"),NULL,LABEL_NORMAL|LAYOUT_RIGHT);
-	combo=sampleFormatComboBox=new FXComboBox(main,25,10,NULL,0,COMBOBOX_STATIC|FRAME_SUNKEN|FRAME_THICK);
+	combo=sampleFormatComboBox=new FXComboBox(main,25,NULL,0,COMBOBOX_STATIC|FRAME_SUNKEN|FRAME_THICK);
+		combo->setNumVisible(10);
 		combo->appendItem(_("8bit Signed PCM"));
 		combo->appendItem(_("8bit Unsigned PCM"));
 		combo->appendItem(_("16bit Signed PCM"));
--- rezound-0.9.0beta/src/frontend_fox/CVoxDialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CVoxDialog.cpp	2004-02-20 01:18:26.878802392 +0100
@@ -45,13 +45,15 @@
 	FXComboBox *combo;
 
 	new FXLabel(main,_("Channels:"));
-	combo=channelsCountComboBox=new FXComboBox(main,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+	combo=channelsCountComboBox=new FXComboBox(main,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+		combo->setNumVisible(8);
 		for(unsigned t=1;t<=MAX_CHANNELS;t++)
 			combo->appendItem(istring(t).c_str());
 		combo->setCurrentItem(1); // stereo
 
 	new FXLabel(main,_("Sample Rate:"));
-	combo=sampleRateComboBox=new FXComboBox(main,10,9,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+	combo=sampleRateComboBox=new FXComboBox(main,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK);
+		combo->setNumVisible(9);
 		combo->appendItem("4000");
 		combo->appendItem("8000");
 		combo->appendItem("11025");
--- rezound-0.9.0beta/src/frontend_fox/COggDialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/COggDialog.cpp	2004-02-20 01:19:57.836974648 +0100
@@ -73,7 +73,8 @@
 	CBRButton=new FXRadioButton(main,_("Constant Bit Rate"),this,ID_WHICH_BUTTON);
 		CBRFrame=new FXMatrix(main,2,MATRIX_BY_COLUMNS|FRAME_RAISED|LAYOUT_FILL_X);
 			new FXLabel(CBRFrame,_("Bit Rate: "));
-			bitRateComboBox=new FXComboBox(CBRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			bitRateComboBox=new FXComboBox(CBRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			bitRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(bitRateComboBox);
 
 
@@ -81,13 +82,16 @@
 	VBRButton=new FXRadioButton(main,_("Variable Bit Rate"),this,ID_WHICH_BUTTON);
 		VBRFrame=new FXMatrix(main,2,MATRIX_BY_COLUMNS|FRAME_RAISED|LAYOUT_FILL_X);
 			new FXLabel(VBRFrame,_("Minimum Bit Rate: "));
-			minRateComboBox=new FXComboBox(VBRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			minRateComboBox=new FXComboBox(VBRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			minRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(minRateComboBox);
 			new FXLabel(VBRFrame,_("Normal Bit Rate: "));
-			normRateComboBox=new FXComboBox(VBRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			normRateComboBox=new FXComboBox(VBRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			normRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(normRateComboBox);
 			new FXLabel(VBRFrame,_("Maximum Bit Rate: "));
-			maxRateComboBox=new FXComboBox(VBRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			maxRateComboBox=new FXComboBox(VBRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			maxRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(maxRateComboBox);
 
 
--- rezound-0.9.0beta/src/frontend_fox/CMp3Dialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CMp3Dialog.cpp	2004-02-20 01:22:36.319881584 +0100
@@ -75,7 +75,8 @@
 	CBRButton=new FXRadioButton(main,_("Constant Bit Rate"),this,ID_WHICH_BUTTON);
 		CBRFrame=new FXMatrix(main,2,MATRIX_BY_COLUMNS|FRAME_RAISED|LAYOUT_FILL_X);
 			new FXLabel(CBRFrame,_("Bit Rate: "));
-			bitRateComboBox=new FXComboBox(CBRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			bitRateComboBox=new FXComboBox(CBRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			bitRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(bitRateComboBox);
 
 
@@ -83,13 +84,16 @@
 	ABRButton=new FXRadioButton(main,_("Average Bit Rate"),this,ID_WHICH_BUTTON);
 		ABRFrame=new FXMatrix(main,2,MATRIX_BY_COLUMNS|FRAME_RAISED|LAYOUT_FILL_X);
 			new FXLabel(ABRFrame,_("Minimum Bit Rate: "));
-			minRateComboBox=new FXComboBox(ABRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			minRateComboBox=new FXComboBox(ABRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			minRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(minRateComboBox);
 			new FXLabel(ABRFrame,_("Average Bit Rate: "));
-			normRateComboBox=new FXComboBox(ABRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			normRateComboBox=new FXComboBox(ABRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			normRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(normRateComboBox,true);
 			new FXLabel(ABRFrame,_("Maximum Bit Rate: "));
-			maxRateComboBox=new FXComboBox(ABRFrame,10,8,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			maxRateComboBox=new FXComboBox(ABRFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			maxRateComboBox->setNumVisible(8);
 			fillBitRateComboBox(maxRateComboBox);
 
 
@@ -97,7 +101,8 @@
 	qualityButton=new FXRadioButton(main,_("Variable Bit Rate Quality Setting"),this,ID_WHICH_BUTTON);
 		qualityFrame=new FXMatrix(main,3,MATRIX_BY_COLUMNS|FRAME_RAISED|LAYOUT_FILL_X);
 			new FXLabel(qualityFrame,_("Quality: "));
-			qualityComboBox=new FXComboBox(qualityFrame,10,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			qualityComboBox=new FXComboBox(qualityFrame,10,NULL,0,COMBOBOX_NORMAL|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+			qualityComboBox->setNumVisible(10);
 			for(int t=0;t<10;t++)
 				qualityComboBox->appendItem(istring(t).c_str());
 			qualityComboBox->setCurrentItem(4);
--- rezound-0.9.0beta/src/frontend_fox/CJACKPortChoiceDialog.cpp.orig	2003-06-27 01:24:46.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/CJACKPortChoiceDialog.cpp	2004-02-20 01:23:22.376879856 +0100
@@ -40,7 +40,8 @@
 	FXModalDialogBox(mainWindow,N_("JACK Port Selection"),0,0,FXModalDialogBox::ftVertical)
 {
 	messageLabel=new FXLabel(getFrame(),"",NULL,LAYOUT_CENTER_X);
-	portNamesComboBox=new FXComboBox(getFrame(),10,8,NULL,0,COMBOBOX_NORMAL|LAYOUT_CENTER_X|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+	portNamesComboBox=new FXComboBox(getFrame(),10,NULL,0,COMBOBOX_NORMAL|LAYOUT_CENTER_X|FRAME_SUNKEN|FRAME_THICK|COMBOBOX_STATIC);
+	portNamesComboBox->setNumVisible(8);
 }
 
 CJACKPortChoiceDialog::~CJACKPortChoiceDialog()
--- rezound-0.9.0beta/src/frontend_fox/FXLFOParamValue.cpp.orig	2003-10-27 21:02:19.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/FXLFOParamValue.cpp	2004-02-20 01:34:24.230262848 +0100
@@ -77,13 +77,14 @@
 			N_("Phase")
 		)),
 
-	LFOTypeComboBox(new FXListBox(this,16,this,ID_LFO_TYPE_COMBOBOX,FRAME_SUNKEN|FRAME_THICK|LISTBOX_NORMAL|LAYOUT_CENTER_X|LAYOUT_FIX_WIDTH,0,0,180,0)),
+	LFOTypeComboBox(new FXListBox(this,this,ID_LFO_TYPE_COMBOBOX,FRAME_SUNKEN|FRAME_THICK|LISTBOX_NORMAL|LAYOUT_CENTER_X|LAYOUT_FIX_WIDTH,0,0,180,0)),
 
 	textFont(getApp()->getNormalFont()),
 
 	minWidth(0),
 	minHeight(236)
 {
+	LFOTypeComboBox->setNumVisible(16);
 	amplitudeSlider->setMinSize(0,0);
 	frequencySlider->setMinSize(0,0);
 	phaseSlider->setMinSize(0,0);
--- rezound-0.9.0beta/src/frontend_fox/FXComboTextParamValue.cpp.orig	2003-10-15 23:17:34.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/FXComboTextParamValue.cpp	2004-02-20 01:29:28.448228552 +0100
@@ -53,10 +53,11 @@
 	isEditable(_isEditable),
 
 	titleLabel(new FXLabel(this,gettext(_name),NULL,LABEL_NORMAL|LAYOUT_CENTER_Y)),
-	valueComboBox(new FXComboBox(this,8,min((size_t)items.size(),(size_t)8),this,ID_COMBOBOX, COMBOBOX_NORMAL|(!isEditable ? COMBOBOX_STATIC : 0) | FRAME_SUNKEN|FRAME_THICK | LAYOUT_CENTER_Y|LAYOUT_FILL_X)),
+	valueComboBox(new FXComboBox(this,8,this,ID_COMBOBOX, COMBOBOX_NORMAL|(!isEditable ? COMBOBOX_STATIC : 0) | FRAME_SUNKEN|FRAME_THICK | LAYOUT_CENTER_Y|LAYOUT_FILL_X)),
 
 	textFont(getApp()->getNormalFont())
 {
+	valueComboBox->setNumVisible(min((size_t)items.size(),(size_t)8));
 	// create a smaller font to use 
         FXFontDesc d;
         textFont->getFontDesc(d);
--- rezound-0.9.0beta/src/frontend_fox/FXPluginRoutingParamValue.cpp.orig	2003-11-05 18:01:57.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/FXPluginRoutingParamValue.cpp	2004-02-20 01:36:23.571120280 +0100
@@ -199,7 +199,8 @@
 				new FXButton(p2,_("Clear"),NULL,this,ID_RESET_BUTTON,BUTTON_NORMAL|LAYOUT_FILL_X);
 			p2=new FXPacker(p1,FRAME_NONE | LAYOUT_FILL_X|LAYOUT_FILL_Y);
 				p3=new FXPacker(p2,LAYOUT_FILL_X|LAYOUT_FILL_Y | FRAME_SUNKEN|FRAME_THICK, 0,0,0,0, 0,0,0,0, 0,0);
-					soundList=new FXList(p3,5,NULL,0,LIST_NORMAL|LIST_SINGLESELECT | LAYOUT_FILL_Y|LAYOUT_FILL_X);
+					soundList=new FXList(p3,NULL,0,LIST_NORMAL|LIST_SINGLESELECT | LAYOUT_FILL_Y|LAYOUT_FILL_X);
+					soundList->setNumVisible(5);
 	}
 
 	canvasParent->setBackColor(COL_BACKGROUND);
@@ -1064,7 +1065,8 @@
 			FXPacker *p=new FXHorizontalFrame(source,FRAME_NONE|LAYOUT_FILL_X, 0,0,0,0, 2,2,0,0, 0,0);
 			(new FXLabel(p,_("Use ")))->setFont(textFont);
 
-			FXComboBox *c=new FXComboBox(p,0,2,NULL,0,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN | LAYOUT_FILL_X);
+			FXComboBox *c=new FXComboBox(p,0,NULL,0,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN | LAYOUT_FILL_X);
+			c->setNumVisible(2);
 				// these have to maintain the order in CPluginMapping::RInputDesc::WhenDataRunsOut because the index ends up being the value of that enum
 			c->appendItem(_("Selection"));
 			c->appendItem(_("All"));
@@ -1076,7 +1078,8 @@
 		{
 			FXHorizontalFrame *nodeParent=new FXHorizontalFrame(source,FRAME_NONE|LAYOUT_FILL_X, 0,0,0,0, 2,2,0,0, 0,0);
 
-			FXComboBox *wdroControl=new FXComboBox(nodeParent,8,2,NULL,0,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN);
+			FXComboBox *wdroControl=new FXComboBox(nodeParent,8,NULL,0,COMBOBOX_NORMAL|COMBOBOX_STATIC | FRAME_SUNKEN);
+			wdroControl->setNumVisible(2);
 			wdroControl->setFont(textFont);
 			wdroControl->setTipText(_("When Data Runs Out ..."));
 				// these have to maintain the order in CPluginMapping::RInputDesc::WhenDataRunsOut because the index ends up being the value of that enum
--- rezound-0.9.0beta/src/frontend_fox/Makefile.am.orig	2003-10-25 08:35:16.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/Makefile.am	2004-06-19 22:53:23.352629744 +0200
@@ -107,14 +107,14 @@
 	$(top_builddir)/src/misc/CNestedDataFile/libmiscCN.la \
 	$(top_builddir)/src/PoolFile/libPoolFile.la
 
-rezound_LDADD=$(_rezound_LDADD) @FOX_1_1_LIBS@ @FOX_LIBS@
+rezound_LDADD=$(_rezound_LDADD) @FOX_1_2_LIBS@ @FOX_LIBS@
 
 ## if I don't do this, automake leaves out @abc@ vars when creating the rezound_DEPENDENCES value
 ## so changes to the LADSPA lib (that may or may not be built) don't cause a re-link
 rezound_DEPENDENCIES=$(_rezound_LDADD)
 
 
-AM_CXXFLAGS=@FOX_1_1_CXXFLAGS@ @FOX_CXXFLAGS@
+AM_CXXFLAGS=@FOX_1_2_CXXFLAGS@ @FOX_CXXFLAGS@
 
 noinst_HEADERS=\
 	fox_compat.h \
--- rezound-0.9.0beta/src/frontend_fox/fox_compat.h.orig	2003-10-28 17:02:06.000000000 +0100
+++ rezound-0.9.0beta/src/frontend_fox/fox_compat.h	2004-06-19 22:54:00.872925792 +0200
@@ -6,9 +6,9 @@
 #if defined(HAVE_LIBFOX)
 	#include <fox/fx.h>
 	#include <fox/fxkeys.h>
-#elif defined(HAVE_LIBFOX_1_1)
-	#include <fox-1.1/fx.h>
-	#include <fox-1.1/fxkeys.h>
+#elif defined(HAVE_LIBFOX_1_2)
+	#include <fox-1.2/fx.h>
+	#include <fox-1.2/fxkeys.h>
 #else
 	#error no HAVE_LIBFOX defined
 #endif
--- rezound-0.9.0beta/configure.ac.orig	2004-06-19 22:44:56.095744000 +0200
+++ rezound-0.9.0beta/configure.ac	2004-06-19 22:52:35.806857800 +0200
@@ -574,14 +574,14 @@
 # This is a check for a class in a C++ library. To pinpoint the specific version we need we could change what class is linked.
 # it sets $1_CXXFLAGS and $1_LIBS to either something useful or nothing
 	# need to pass something that says not to error if it's not found so we can continue looking for just libFOX
-ajv_CXX_CHECK_LIB(FOX-1.1,FXApp,fox-1.1/fx.h,[http://fox-toolkit.org],[],false)
-if test -z "$FOX_1_1_LIBS"; then
-	# libFOX-1.1 was not found, now try libFOX
+ajv_CXX_CHECK_LIB(FOX-1.2,FXApp,fox-1.2/fx.h,[http://fox-toolkit.org],[],false)
+if test -z "$FOX_1_2_LIBS"; then
+	# libFOX-1.2 was not found, now try libFOX
 	ajv_CXX_CHECK_LIB(FOX,FXApp,fox/fx.h,[http://fox-toolkit.org],[],true)
 fi
 
-AC_SUBST(FOX_1_1_CXXFLAGS)
-AC_SUBST(FOX_1_1_LIBS)
+AC_SUBST(FOX_1_2_CXXFLAGS)
+AC_SUBST(FOX_1_2_LIBS)
 AC_SUBST(FOX_CXXFLAGS)
 AC_SUBST(FOX_LIBS)
 
@@ -645,7 +645,7 @@
 		AC_MSG_NOTICE([*** NOTICE *** stand-alone build enabled -- now static versions of all the dependancy libraries will be required.  If final linking fails, then make sure a static version of the failed library is installed.])
 
 		# warn about needing static libs if using FOX and statically linking
-		if test "$FOX_LIBS" != "" -o "$FOX_1_1_LIBS" != "" ; then
+		if test "$FOX_LIBS" != "" -o "$FOX_1_2_LIBS" != "" ; then
 			AC_MSG_NOTICE([*** NOTICE *** since static linking is enabled, FOX will require static versions of libjpeg, libpng, libtiff, XFree86-libs, and maybe others (Also, I have found it necessary to configure FOX as: "CPPFLAGS=-DNO_XIM configure --with-opengl=no" when statically linking)])
 		fi
 	]
--- rezound-0.9.0beta/src/frontend_fox/FXPluginRoutingParamValue.h.orig	2003-10-22 07:11:24.000000000 +0200
+++ rezound-0.9.0beta/src/frontend_fox/FXPluginRoutingParamValue.h	2004-06-20 11:12:14.412284296 +0200
@@ -24,8 +24,6 @@
 #include "../../config/common.h"
 #include "fox_compat.h"
 
-#include <fox/fx.h>
-
 #include <string>
 #include <vector>
 #include <utility>
