diff -aurN rezound-0.8.1beta.orig/src/misc/CPath.h rezound-0.8.1beta/src/misc/CPath.h
--- rezound-0.8.1beta.orig/src/misc/CPath.h	2003-04-08 06:00:07.000000000 +0200
+++ rezound-0.8.1beta/src/misc/CPath.h	2003-06-27 21:23:40.000000000 +0200
@@ -71,7 +71,8 @@
 
 	CPath(const string &_path="") :
 		path(NULL),
-		_exists(false)
+		_exists(false),
+		_permDenied(false)
 	{
 		setPath(_path);
 	}
@@ -96,11 +97,15 @@
 		if(stat(path,&statBuf)!=0)
 		{
 			_exists=false;
-			if(errno!=ENOENT)
+			_permDenied= (errno==EACCES);
+			if(errno!=ENOENT && errno!=EACCES)
 				throw runtime_error(string(__func__)+" -- error stat-ing path name -- '"+path+"' -- "+strerror(errno));
 		}
 		else
+		{
 			_exists=true;
+			_permDenied=false;
+		}
 	}
 
 	/*
@@ -113,6 +118,7 @@
 
 	/*
 		returns true iff the pathname exists and we could access it
+		hence, permission denied makes this returned false even if it did exist :-/
 	*/
 	bool exists() const
 	{
@@ -120,6 +126,14 @@
 	}
 
 	/*
+		returns true if the maybe some of the pathname exists but we couldn't access it
+	*/
+	bool permDenied() const
+	{
+		return(_permDenied);
+	}
+
+	/*
 		possbily creates the path and/or updates its time(s) to current
 	*/
 	bool touch(bool canCreate=true,bool throwIfError=true) const
@@ -254,6 +268,7 @@
 private:
 	char *path;
 	bool _exists;
+	bool _permDenied;
 	struct stat statBuf;
 
 };
